crates/ty_ide/src/hover.rs

```diff
@@ -165,6 +182,261 @@ mod tests {
+            b: coming for `a`'s title
+
+        ```
+        ---------------------------------------------
+        info[hover]: Hovered content is
+          --> main.py:11:9
+           |
+         9 |             return 0
+        10 |
+        11 |         my_func(1, 2)
+           |         ^^^^^-^
+           |         |    |
+           |         |    Cursor offset
+           |         source
+           |
+        ");
+    }
+
+    #[test]
+    fn hover_class() {
+        let test = cursor_test(
+            r#"
+        class MyClass:
+            '''
+                This is such a great class!!
+
+                    Don't you know?
+                
+                Everyone loves my class!!
+
+            '''
+            def __init__(self, val):
+                """initializes MyClass (perfectly)"""
+                self.val = val
+            
+            def my_method(self, a, b):
+                '''This is such a great func!!
+
+                Args:
+                    a: first for a reason
+                    b: coming for `a`'s title
+                '''
+                return 0
+
+        MyCla<CURSOR>ss
+        "#,
+        );
+
+        assert_snapshot!(test.hover(), @r"
+        <class 'MyClass'>
+        ---------------------------------------------
+        This is such a great class!!
+
+            Don't you know?
+
+        Everyone loves my class!!
+
+        ---------------------------------------------
+        ```python
+        <class 'MyClass'>
+        ```
+        ---
+        ```text
+        This is such a great class!!
+
+            Don't you know?
+
+        Everyone loves my class!!
+
+        ```
+        ---------------------------------------------
+        info[hover]: Hovered content is
+          --> main.py:24:9
+           |
+        22 |                 return 0
+        23 |
+        24 |         MyClass
+           |         ^^^^^-^
+           |         |    |
+           |         |    Cursor offset
+           |         source
+           |
+        ");
+    }
+
+    #[test]
+    fn hover_class_init() {
+        let test = cursor_test(
+            r#"
+        class MyClass:
+            '''
+                This is such a great class!!
+
+                    Don't you know?
+                
+                Everyone loves my class!!
+
+            '''
+            def __init__(self, val):
+                """initializes MyClass (perfectly)"""
+                self.val = val
+            
+            def my_method(self, a, b):
+                '''This is such a great func!!
+
+                Args:
+                    a: first for a reason
+                    b: coming for `a`'s title
+                '''
+                return 0
+
+        x = MyCla<CURSOR>ss(0)
+        "#,
+        );
+
+        assert_snapshot!(test.hover(), @r"
+        <class 'MyClass'>
+        ---------------------------------------------
+        This is such a great class!!
```

Comment:
This result is a disappointment but a pre-existing issue that we don't really distinguish a class from an invocation of the initializer (future work).